pkg/backtest/matching.go

```diff
@@ -147,6 +147,11 @@ func (m *SimplePriceMatching) PlaceOrder(o types.SubmitOrder) (*types.Order, *ty
 
 	case types.OrderTypeLimit, types.OrderTypeStopLimit, types.OrderTypeLimitMaker:
 		o.Price = m.Market.TruncatePrice(o.Price)
+		// if o.Price makes the limit order a taker, use the latest price as the order price
+		// otherwise we'll endup locking more assets than needed
+		if isLimitTakerOrder(o, m.LastPrice) {
+			o.Price = m.LastPrice
```

Comment:
maybe you can assign to AveragePrice? the price of the limit order can not be changed.